src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -29,19 +27,16 @@
 	}
 	
 	@GetMapping
-	public ResponseEntity<Object> findByAuthor(@RequestParam String author) {
+	public Book findByAuthor(@RequestParam String author) {
 		System.out.println("En BookController -> findByAuthor");
 		//Sorting with a Sort Parameter 
 		/*
 		Sort sortByYear = Sort.by("year").descending();
 		Book book bookRepository.findByAuthorAndSort(author, sortByYear);
 		*/
 		//Sorting With the OrderBy Method Keyword
-		Book book = bookRepository.findFirstByAuthorOrderByYearDesc(author);
-		if (book == null) {
-			return ResponseEntity.notFound().build();
-		} else {			
-			return ResponseEntity.ok(book);
-		}
+		Optional<Book> optionalBook = bookRepository.findFirstByAuthorOrderByYear(author);		
+		return optionalBook.orElseThrow(() -> new BookNotFoundException("No se encontro libro del autor " + author));
```

Comment:
Genial! Un detalle más y estamos! Acá no es necesario tener una variable temporal `optionalBook` ya que no la estamos trabajando y tampoco aclara tanto el código, podría reescribirse a:
```
return bookRepository.findFirstByAuthorOrderByYear(author)
		.orElseThrow(() -> new BookNotFoundException("No se encontro libro del autor " + author));
```

En general se usan de esta manera los Optional, de esta manera queda más declarativo ("Buscar el primer libro por autor y sino arrojar una book not found exception")

Reply:
Variable eliminada.